internal/indexer/walker.go

```diff
@@ -16,72 +16,60 @@ func (idx *Indexer) WalkedModule(ctx context.Context, modHandle document.DirHand
 	ids := make(job.IDs, 0)
 	var errs *multierror.Error
 
-	// blockingJobIds tracks job IDs which need to finish
-	// prior to collecting references
-	blockingJobIds := make(job.IDs, 0)
-
-	id, err := idx.jobStore.EnqueueJob(job.Job{
+	parseId, err := idx.jobStore.EnqueueJob(job.Job{
 		Dir: modHandle,
 		Func: func(ctx context.Context) error {
 			return module.ParseModuleConfiguration(idx.fs, idx.modStore, modHandle.Path())
 		},
 		Type: op.OpTypeParseModuleConfiguration.String(),
-		Defer: func(ctx context.Context, jobErr error) (job.IDs, error) {
-			ids := make(job.IDs, 0)
-
-			id, err := idx.jobStore.EnqueueJob(job.Job{
-				Dir:  modHandle,
-				Type: op.OpTypeLoadModuleMetadata.String(),
-				Func: func(ctx context.Context) error {
-					return module.LoadModuleMetadata(idx.modStore, modHandle.Path())
-				},
-			})
-			if err != nil {
-				return ids, err
-			} else {
-				ids = append(ids, id)
-			}
+	})
+	if err != nil {
+		errs = multierror.Append(errs, err)
```

Comment:
Are we able to continue in the case of an error here? We're using `parseId` in the next job's `DependsOn`, but `parseId` is likely to be `""`.

Reply:
Good catch - passing empty string to `DependsOn` is certainly unwanted here! 🙈